From 42ccde7bc21b270a118e4adebef36c99200e3d7f Mon Sep 17 00:00:00 2001
From: John Audia <therealgraysky@proton.me>
Date: Wed, 13 Dec 2023 14:52:38 -0500
Subject: [PATCH] kernel: bump 6.1 to 6.1.68

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.1.68

Removed upstreamed:
	generic/backport-6.1/795-v6.6-12-r8152-Rename-RTL8152_UNPLUG-to-RTL8152_INACCESSIBLE.patch[1]

Manually rebased:
	mediatek/patches-6.1/100-dts-update-mt7622-rfb1.patch

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.68&id=3759e735562a31e44fee825498f05c06e64b25a8

Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

Signed-off-by: John Audia <therealgraysky@proton.me>
---
 include/kernel-6.1                            |   4 +-
 .../900-unaligned_access_hacks.patch          |   2 +-
 ...-Insert-mb-for-correct-FIFO-handling.patch |   2 +-
 ...Add-cts-event-workaround-DT-property.patch |   2 +-
 ...011-Avoid-rare-write-when-full-error.patch |   2 +-
 ...is7xx-Don-t-spin-if-no-data-received.patch |   4 +-
 ...B-device-driver-for-config-selection.patch |   6 +-
 ...-avoid-to-change-cfg-for-all-devices.patch |   6 +-
 ...8152-remove-rtl_vendor_mode-function.patch |   4 +-
 ...-control-transfer-of-rtl8152_get_ver.patch |   2 +-
 ...152-fix-the-autosuspend-doesn-t-work.patch |   2 +-
 .../795-v6.6-09-r8152-set-bp-in-bulk.patch    |   8 +-
 ...eth-r8152-try-to-use-a-normal-budget.patch |   2 +-
 ...r-device-ID-pair-for-D-Link-DUB-E250.patch |   2 +-
 ...L8152_UNPLUG-to-RTL8152_INACCESSIBLE.patch | 447 ------------------
 ...re-register-access-if-register-acces.patch |  14 +-
 ...-r8152-add-LED-configuration-from-OF.patch |   8 +-
 .../800-GPIO-add-named-gpio-exports.patch     |   6 +-
 ...s-negative-stack-offsets-on-stack-tr.patch |   2 +-
 .../300-mips_expose_boot_raw.patch            |   4 +-
 ...les-ignore-EOPNOTSUPP-on-flowtable-d.patch |   2 +-
 ...c-move-TX-timer-arm-after-DMA-enable.patch |  12 +-
 ...et-stmmac-improve-TX-timer-arm-logic.patch |   2 +-
 .../100-dts-update-mt7622-rfb1.patch          |   3 +-
 .../100-mips_image_cmdline_hack.patch         |   2 +-
 25 files changed, 52 insertions(+), 498 deletions(-)
 delete mode 100644 target/linux/generic/backport-6.1/795-v6.6-12-r8152-Rename-RTL8152_UNPLUG-to-RTL8152_INACCESSIBLE.patch

diff --git a/include/kernel-6.1 b/include/kernel-6.1
index ef5b2b0b6ef7c..d84b0b21c4364 100644
--- a/include/kernel-6.1
+++ b/include/kernel-6.1
@@ -1,2 +1,2 @@
-LINUX_VERSION-6.1 = .67
-LINUX_KERNEL_HASH-6.1.67 = 7537db7289ca4854a126bc1237c47c5b21784bcbf27b4e571d389e3528c59285
+LINUX_VERSION-6.1 = .68
+LINUX_KERNEL_HASH-6.1.68 = 365ff26a30e206de9b18489f45d38582a0a61b7c5919f8ab89295a47316784e1
diff --git a/target/linux/ath79/patches-6.1/900-unaligned_access_hacks.patch b/target/linux/ath79/patches-6.1/900-unaligned_access_hacks.patch
index 5bd7d0ba635b6..b884b8334bdd8 100644
--- a/target/linux/ath79/patches-6.1/900-unaligned_access_hacks.patch
+++ b/target/linux/ath79/patches-6.1/900-unaligned_access_hacks.patch
@@ -750,7 +750,7 @@ SVN-Revision: 35130
  EXPORT_SYMBOL(xfrm_parse_spi);
 --- a/net/ipv4/tcp_input.c
 +++ b/net/ipv4/tcp_input.c
-@@ -4179,14 +4179,16 @@ static bool tcp_parse_aligned_timestamp(
+@@ -4183,14 +4183,16 @@ static bool tcp_parse_aligned_timestamp(
  {
  	const __be32 *ptr = (const __be32 *)(th + 1);
  
diff --git a/target/linux/bcm27xx/patches-6.1/950-0101-amba_pl011-Insert-mb-for-correct-FIFO-handling.patch b/target/linux/bcm27xx/patches-6.1/950-0101-amba_pl011-Insert-mb-for-correct-FIFO-handling.patch
index 9fd8339530a5b..67a9eb97454d9 100644
--- a/target/linux/bcm27xx/patches-6.1/950-0101-amba_pl011-Insert-mb-for-correct-FIFO-handling.patch
+++ b/target/linux/bcm27xx/patches-6.1/950-0101-amba_pl011-Insert-mb-for-correct-FIFO-handling.patch
@@ -17,7 +17,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
 
 --- a/drivers/tty/serial/amba-pl011.c
 +++ b/drivers/tty/serial/amba-pl011.c
-@@ -1430,6 +1430,7 @@ static bool pl011_tx_char(struct uart_am
+@@ -1426,6 +1426,7 @@ static bool pl011_tx_char(struct uart_am
  		return false; /* unable to transmit character */
  
  	pl011_write(c, uap, REG_DR);
diff --git a/target/linux/bcm27xx/patches-6.1/950-0102-amba_pl011-Add-cts-event-workaround-DT-property.patch b/target/linux/bcm27xx/patches-6.1/950-0102-amba_pl011-Add-cts-event-workaround-DT-property.patch
index a67aa16259b7d..25386a174cba0 100644
--- a/target/linux/bcm27xx/patches-6.1/950-0102-amba_pl011-Add-cts-event-workaround-DT-property.patch
+++ b/target/linux/bcm27xx/patches-6.1/950-0102-amba_pl011-Add-cts-event-workaround-DT-property.patch
@@ -36,7 +36,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
    - reg
 --- a/drivers/tty/serial/amba-pl011.c
 +++ b/drivers/tty/serial/amba-pl011.c
-@@ -2805,6 +2805,11 @@ static int pl011_probe(struct amba_devic
+@@ -2801,6 +2801,11 @@ static int pl011_probe(struct amba_devic
  	if (IS_ERR(uap->clk))
  		return PTR_ERR(uap->clk);
  
diff --git a/target/linux/bcm27xx/patches-6.1/950-0103-tty-amba-pl011-Avoid-rare-write-when-full-error.patch b/target/linux/bcm27xx/patches-6.1/950-0103-tty-amba-pl011-Avoid-rare-write-when-full-error.patch
index 08bf4098e3bae..8a0a835d27159 100644
--- a/target/linux/bcm27xx/patches-6.1/950-0103-tty-amba-pl011-Avoid-rare-write-when-full-error.patch
+++ b/target/linux/bcm27xx/patches-6.1/950-0103-tty-amba-pl011-Avoid-rare-write-when-full-error.patch
@@ -29,7 +29,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
 
 --- a/drivers/tty/serial/amba-pl011.c
 +++ b/drivers/tty/serial/amba-pl011.c
-@@ -1491,6 +1491,10 @@ static bool pl011_tx_chars(struct uart_a
+@@ -1487,6 +1487,10 @@ static bool pl011_tx_chars(struct uart_a
  		if (likely(from_irq) && count-- == 0)
  			break;
  
diff --git a/target/linux/bcm27xx/patches-6.1/950-0146-sc16is7xx-Don-t-spin-if-no-data-received.patch b/target/linux/bcm27xx/patches-6.1/950-0146-sc16is7xx-Don-t-spin-if-no-data-received.patch
index e4e4909510d4d..30fcdb5218805 100644
--- a/target/linux/bcm27xx/patches-6.1/950-0146-sc16is7xx-Don-t-spin-if-no-data-received.patch
+++ b/target/linux/bcm27xx/patches-6.1/950-0146-sc16is7xx-Don-t-spin-if-no-data-received.patch
@@ -12,8 +12,8 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.org>
 
 --- a/drivers/tty/serial/sc16is7xx.c
 +++ b/drivers/tty/serial/sc16is7xx.c
-@@ -771,6 +771,8 @@ static bool sc16is7xx_port_irq(struct sc
- 			rxlen = sc16is7xx_port_read(port, SC16IS7XX_RXLVL_REG);
+@@ -783,6 +783,8 @@ static bool sc16is7xx_port_irq(struct sc
+ 
  			if (rxlen)
  				sc16is7xx_handle_rx(port, rxlen, iir);
 +			else
diff --git a/target/linux/generic/backport-6.1/795-v6.3-01-r8152-add-USB-device-driver-for-config-selection.patch b/target/linux/generic/backport-6.1/795-v6.3-01-r8152-add-USB-device-driver-for-config-selection.patch
index 605faeec035df..8104ed28df849 100644
--- a/target/linux/generic/backport-6.1/795-v6.3-01-r8152-add-USB-device-driver-for-config-selection.patch
+++ b/target/linux/generic/backport-6.1/795-v6.3-01-r8152-add-USB-device-driver-for-config-selection.patch
@@ -87,7 +87,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -9661,6 +9661,9 @@ static int rtl8152_probe(struct usb_inte
+@@ -9675,6 +9675,9 @@ static int rtl8152_probe(struct usb_inte
  	if (version == RTL_VER_UNKNOWN)
  		return -ENODEV;
  
@@ -97,7 +97,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	if (!rtl_vendor_mode(intf))
  		return -ENODEV;
  
-@@ -9861,43 +9864,35 @@ static void rtl8152_disconnect(struct us
+@@ -9875,43 +9878,35 @@ static void rtl8152_disconnect(struct us
  	}
  }
  
@@ -164,7 +164,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	{}
  };
  
-@@ -9917,7 +9912,61 @@ static struct usb_driver rtl8152_driver
+@@ -9931,7 +9926,61 @@ static struct usb_driver rtl8152_driver
  	.disable_hub_initiated_lpm = 1,
  };
  
diff --git a/target/linux/generic/backport-6.1/795-v6.3-03-r8152-avoid-to-change-cfg-for-all-devices.patch b/target/linux/generic/backport-6.1/795-v6.3-03-r8152-avoid-to-change-cfg-for-all-devices.patch
index a5b01f7b0829a..6476a7fe67e11 100644
--- a/target/linux/generic/backport-6.1/795-v6.3-03-r8152-avoid-to-change-cfg-for-all-devices.patch
+++ b/target/linux/generic/backport-6.1/795-v6.3-03-r8152-avoid-to-change-cfg-for-all-devices.patch
@@ -17,7 +17,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -9542,9 +9542,8 @@ static int rtl_fw_init(struct r8152 *tp)
+@@ -9556,9 +9556,8 @@ static int rtl_fw_init(struct r8152 *tp)
  	return 0;
  }
  
@@ -28,7 +28,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	u32 ocp_data = 0;
  	__le32 *tmp;
  	u8 version;
-@@ -9614,10 +9613,19 @@ u8 rtl8152_get_version(struct usb_interf
+@@ -9628,10 +9627,19 @@ u8 rtl8152_get_version(struct usb_interf
  		break;
  	default:
  		version = RTL_VER_UNKNOWN;
@@ -49,7 +49,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	dev_dbg(&intf->dev, "Detected version 0x%04x\n", version);
  
  	return version;
-@@ -9917,6 +9925,12 @@ static int rtl8152_cfgselector_probe(str
+@@ -9931,6 +9939,12 @@ static int rtl8152_cfgselector_probe(str
  	struct usb_host_config *c;
  	int i, num_configs;
  
diff --git a/target/linux/generic/backport-6.1/795-v6.3-04-r8152-remove-rtl_vendor_mode-function.patch b/target/linux/generic/backport-6.1/795-v6.3-04-r8152-remove-rtl_vendor_mode-function.patch
index f97750861c3e4..7e647d0cb56ed 100644
--- a/target/linux/generic/backport-6.1/795-v6.3-04-r8152-remove-rtl_vendor_mode-function.patch
+++ b/target/linux/generic/backport-6.1/795-v6.3-04-r8152-remove-rtl_vendor_mode-function.patch
@@ -16,7 +16,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -8274,43 +8274,6 @@ static bool rtl_check_vendor_ok(struct u
+@@ -8288,43 +8288,6 @@ static bool rtl_check_vendor_ok(struct u
  	return true;
  }
  
@@ -60,7 +60,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  static int rtl8152_pre_reset(struct usb_interface *intf)
  {
  	struct r8152 *tp = usb_get_intfdata(intf);
-@@ -9672,7 +9635,7 @@ static int rtl8152_probe(struct usb_inte
+@@ -9686,7 +9649,7 @@ static int rtl8152_probe(struct usb_inte
  	if (intf->cur_altsetting->desc.bInterfaceClass != USB_CLASS_VENDOR_SPEC)
  		return -ENODEV;
  
diff --git a/target/linux/generic/backport-6.1/795-v6.3-05-r8152-reduce-the-control-transfer-of-rtl8152_get_ver.patch b/target/linux/generic/backport-6.1/795-v6.3-05-r8152-reduce-the-control-transfer-of-rtl8152_get_ver.patch
index 421f2c7f8f51d..565fbb3074f97 100644
--- a/target/linux/generic/backport-6.1/795-v6.3-05-r8152-reduce-the-control-transfer-of-rtl8152_get_ver.patch
+++ b/target/linux/generic/backport-6.1/795-v6.3-05-r8152-reduce-the-control-transfer-of-rtl8152_get_ver.patch
@@ -18,7 +18,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -9624,20 +9624,21 @@ static int rtl8152_probe(struct usb_inte
+@@ -9638,20 +9638,21 @@ static int rtl8152_probe(struct usb_inte
  			 const struct usb_device_id *id)
  {
  	struct usb_device *udev = interface_to_usbdev(intf);
diff --git a/target/linux/generic/backport-6.1/795-v6.4-07-r8152-fix-the-autosuspend-doesn-t-work.patch b/target/linux/generic/backport-6.1/795-v6.4-07-r8152-fix-the-autosuspend-doesn-t-work.patch
index b4d5b8bdb9151..cd245f9d2e8ff 100644
--- a/target/linux/generic/backport-6.1/795-v6.4-07-r8152-fix-the-autosuspend-doesn-t-work.patch
+++ b/target/linux/generic/backport-6.1/795-v6.4-07-r8152-fix-the-autosuspend-doesn-t-work.patch
@@ -14,7 +14,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -9925,6 +9925,7 @@ static struct usb_device_driver rtl8152_
+@@ -9939,6 +9939,7 @@ static struct usb_device_driver rtl8152_
  	.probe =	rtl8152_cfgselector_probe,
  	.id_table =	rtl8152_table,
  	.generic_subclass = 1,
diff --git a/target/linux/generic/backport-6.1/795-v6.6-09-r8152-set-bp-in-bulk.patch b/target/linux/generic/backport-6.1/795-v6.6-09-r8152-set-bp-in-bulk.patch
index 485a005b174bd..bc70c5af02c70 100644
--- a/target/linux/generic/backport-6.1/795-v6.6-09-r8152-set-bp-in-bulk.patch
+++ b/target/linux/generic/backport-6.1/795-v6.6-09-r8152-set-bp-in-bulk.patch
@@ -15,7 +15,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -3984,29 +3984,10 @@ static void rtl_reset_bmu(struct r8152 *
+@@ -3990,29 +3990,10 @@ static void rtl_reset_bmu(struct r8152 *
  /* Clear the bp to stop the firmware before loading a new one */
  static void rtl_clear_bp(struct r8152 *tp, u16 type)
  {
@@ -48,7 +48,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	case RTL_VER_08:
  	case RTL_VER_09:
  	case RTL_VER_10:
-@@ -4014,32 +3995,31 @@ static void rtl_clear_bp(struct r8152 *t
+@@ -4020,32 +4001,31 @@ static void rtl_clear_bp(struct r8152 *t
  	case RTL_VER_12:
  	case RTL_VER_13:
  	case RTL_VER_15:
@@ -100,7 +100,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  
  	/* wait 3 ms to make sure the firmware is stopped */
  	usleep_range(3000, 6000);
-@@ -5016,10 +4996,9 @@ static void rtl8152_fw_phy_nc_apply(stru
+@@ -5022,10 +5002,9 @@ static void rtl8152_fw_phy_nc_apply(stru
  
  static void rtl8152_fw_mac_apply(struct r8152 *tp, struct fw_mac *mac)
  {
@@ -112,7 +112,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  
  	switch (__le32_to_cpu(mac->blk_hdr.type)) {
  	case RTL_FW_PLA:
-@@ -5061,12 +5040,8 @@ static void rtl8152_fw_mac_apply(struct
+@@ -5067,12 +5046,8 @@ static void rtl8152_fw_mac_apply(struct
  	ocp_write_word(tp, type, __le16_to_cpu(mac->bp_ba_addr),
  		       __le16_to_cpu(mac->bp_ba_value));
  
diff --git a/target/linux/generic/backport-6.1/795-v6.6-10-eth-r8152-try-to-use-a-normal-budget.patch b/target/linux/generic/backport-6.1/795-v6.6-10-eth-r8152-try-to-use-a-normal-budget.patch
index 864671bb32f16..d7fdcdb2c6300 100644
--- a/target/linux/generic/backport-6.1/795-v6.6-10-eth-r8152-try-to-use-a-normal-budget.patch
+++ b/target/linux/generic/backport-6.1/795-v6.6-10-eth-r8152-try-to-use-a-normal-budget.patch
@@ -27,7 +27,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -9770,8 +9770,7 @@ static int rtl8152_probe(struct usb_inte
+@@ -9784,8 +9784,7 @@ static int rtl8152_probe(struct usb_inte
  
  	usb_set_intfdata(intf, tp);
  
diff --git a/target/linux/generic/backport-6.1/795-v6.6-11-r8152-add-vendor-device-ID-pair-for-D-Link-DUB-E250.patch b/target/linux/generic/backport-6.1/795-v6.6-11-r8152-add-vendor-device-ID-pair-for-D-Link-DUB-E250.patch
index ab6563d5c4362..e4375963675bc 100644
--- a/target/linux/generic/backport-6.1/795-v6.6-11-r8152-add-vendor-device-ID-pair-for-D-Link-DUB-E250.patch
+++ b/target/linux/generic/backport-6.1/795-v6.6-11-r8152-add-vendor-device-ID-pair-for-D-Link-DUB-E250.patch
@@ -29,7 +29,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  extern u8 rtl8152_get_version(struct usb_interface *intf);
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -9846,6 +9846,7 @@ static const struct usb_device_id rtl815
+@@ -9860,6 +9860,7 @@ static const struct usb_device_id rtl815
  	{ USB_DEVICE(VENDOR_ID_LINKSYS, 0x0041) },
  	{ USB_DEVICE(VENDOR_ID_NVIDIA,  0x09ff) },
  	{ USB_DEVICE(VENDOR_ID_TPLINK,  0x0601) },
diff --git a/target/linux/generic/backport-6.1/795-v6.6-12-r8152-Rename-RTL8152_UNPLUG-to-RTL8152_INACCESSIBLE.patch b/target/linux/generic/backport-6.1/795-v6.6-12-r8152-Rename-RTL8152_UNPLUG-to-RTL8152_INACCESSIBLE.patch
deleted file mode 100644
index 480a60212ae61..0000000000000
--- a/target/linux/generic/backport-6.1/795-v6.6-12-r8152-Rename-RTL8152_UNPLUG-to-RTL8152_INACCESSIBLE.patch
+++ /dev/null
@@ -1,447 +0,0 @@
-From 715f67f33af45ce2cc3a5b1ef133cc8c8e7787b0 Mon Sep 17 00:00:00 2001
-From: Douglas Anderson <dianders@chromium.org>
-Date: Fri, 20 Oct 2023 14:06:58 -0700
-Subject: [PATCH] r8152: Rename RTL8152_UNPLUG to RTL8152_INACCESSIBLE
-
-Whenever the RTL8152_UNPLUG is set that just tells the driver that all
-accesses will fail and we should just immediately bail. A future patch
-will use this same concept at a time when the driver hasn't actually
-been unplugged but is about to be reset. Rename the flag in
-preparation for the future patch.
-
-This is a no-op change and just a search and replace.
-
-Signed-off-by: Douglas Anderson <dianders@chromium.org>
-Reviewed-by: Grant Grundler <grundler@chromium.org>
-Signed-off-by: David S. Miller <davem@davemloft.net>
----
- drivers/net/usb/r8152.c | 96 ++++++++++++++++++++---------------------
- 1 file changed, 48 insertions(+), 48 deletions(-)
-
---- a/drivers/net/usb/r8152.c
-+++ b/drivers/net/usb/r8152.c
-@@ -763,7 +763,7 @@ enum rtl_register_content {
- 
- /* rtl8152 flags */
- enum rtl8152_flags {
--	RTL8152_UNPLUG = 0,
-+	RTL8152_INACCESSIBLE = 0,
- 	RTL8152_SET_RX_MODE,
- 	WORK_ENABLE,
- 	RTL8152_LINK_CHG,
-@@ -1244,7 +1244,7 @@ int set_registers(struct r8152 *tp, u16
- static void rtl_set_unplug(struct r8152 *tp)
- {
- 	if (tp->udev->state == USB_STATE_NOTATTACHED) {
--		set_bit(RTL8152_UNPLUG, &tp->flags);
-+		set_bit(RTL8152_INACCESSIBLE, &tp->flags);
- 		smp_mb__after_atomic();
- 	}
- }
-@@ -1255,7 +1255,7 @@ static int generic_ocp_read(struct r8152
- 	u16 limit = 64;
- 	int ret = 0;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	/* both size and indix must be 4 bytes align */
-@@ -1299,7 +1299,7 @@ static int generic_ocp_write(struct r815
- 	u16 byteen_start, byteen_end, byen;
- 	u16 limit = 512;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	/* both size and indix must be 4 bytes align */
-@@ -1536,7 +1536,7 @@ static int read_mii_word(struct net_devi
- 	struct r8152 *tp = netdev_priv(netdev);
- 	int ret;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	if (phy_id != R8152_PHY_ID)
-@@ -1552,7 +1552,7 @@ void write_mii_word(struct net_device *n
- {
- 	struct r8152 *tp = netdev_priv(netdev);
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	if (phy_id != R8152_PHY_ID)
-@@ -1757,7 +1757,7 @@ static void read_bulk_callback(struct ur
- 	if (!tp)
- 		return;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	if (!test_bit(WORK_ENABLE, &tp->flags))
-@@ -1849,7 +1849,7 @@ static void write_bulk_callback(struct u
- 	if (!test_bit(WORK_ENABLE, &tp->flags))
- 		return;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	if (!skb_queue_empty(&tp->tx_queue))
-@@ -1870,7 +1870,7 @@ static void intr_callback(struct urb *ur
- 	if (!test_bit(WORK_ENABLE, &tp->flags))
- 		return;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	switch (status) {
-@@ -2614,7 +2614,7 @@ static void bottom_half(struct tasklet_s
- {
- 	struct r8152 *tp = from_tasklet(tp, t, tx_tl);
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	if (!test_bit(WORK_ENABLE, &tp->flags))
-@@ -2657,7 +2657,7 @@ int r8152_submit_rx(struct r8152 *tp, st
- 	int ret;
- 
- 	/* The rx would be stopped, so skip submitting */
--	if (test_bit(RTL8152_UNPLUG, &tp->flags) ||
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags) ||
- 	    !test_bit(WORK_ENABLE, &tp->flags) || !netif_carrier_ok(tp->netdev))
- 		return 0;
- 
-@@ -3057,7 +3057,7 @@ static int rtl_enable(struct r8152 *tp)
- 
- static int rtl8152_enable(struct r8152 *tp)
- {
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	set_tx_qlen(tp);
-@@ -3144,7 +3144,7 @@ static int rtl8153_enable(struct r8152 *
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	set_tx_qlen(tp);
-@@ -3176,7 +3176,7 @@ static void rtl_disable(struct r8152 *tp
- 	u32 ocp_data;
- 	int i;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags)) {
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags)) {
- 		rtl_drop_queued_tx(tp);
- 		return;
- 	}
-@@ -3630,7 +3630,7 @@ static u16 r8153_phy_status(struct r8152
- 		}
- 
- 		msleep(20);
--		if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+		if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 			break;
- 	}
- 
-@@ -3662,7 +3662,7 @@ static void r8153b_ups_en(struct r8152 *
- 			int i;
- 
- 			for (i = 0; i < 500; i++) {
--				if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+				if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 					return;
- 				if (ocp_read_word(tp, MCU_TYPE_PLA, PLA_BOOT_CTRL) &
- 				    AUTOLOAD_DONE)
-@@ -3704,7 +3704,7 @@ static void r8153c_ups_en(struct r8152 *
- 			int i;
- 
- 			for (i = 0; i < 500; i++) {
--				if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+				if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 					return;
- 				if (ocp_read_word(tp, MCU_TYPE_PLA, PLA_BOOT_CTRL) &
- 				    AUTOLOAD_DONE)
-@@ -4049,8 +4049,8 @@ static int rtl_phy_patch_request(struct
- 	for (i = 0; wait && i < 5000; i++) {
- 		u32 ocp_data;
- 
--		if (test_bit(RTL8152_UNPLUG, &tp->flags))
--			break;
-+		if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
-+			return -ENODEV;
- 
- 		usleep_range(1000, 2000);
- 		ocp_data = ocp_reg_read(tp, OCP_PHY_PATCH_STAT);
-@@ -6008,7 +6008,7 @@ static int rtl8156_enable(struct r8152 *
- 	u32 ocp_data;
- 	u16 speed;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	r8156_fc_parameter(tp);
-@@ -6066,7 +6066,7 @@ static int rtl8156b_enable(struct r8152
- 	u32 ocp_data;
- 	u16 speed;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	set_tx_qlen(tp);
-@@ -6252,7 +6252,7 @@ out:
- 
- static void rtl8152_up(struct r8152 *tp)
- {
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8152_aldps_en(tp, false);
-@@ -6262,7 +6262,7 @@ static void rtl8152_up(struct r8152 *tp)
- 
- static void rtl8152_down(struct r8152 *tp)
- {
--	if (test_bit(RTL8152_UNPLUG, &tp->flags)) {
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags)) {
- 		rtl_drop_queued_tx(tp);
- 		return;
- 	}
-@@ -6277,7 +6277,7 @@ static void rtl8153_up(struct r8152 *tp)
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153_u1u2en(tp, false);
-@@ -6317,7 +6317,7 @@ static void rtl8153_down(struct r8152 *t
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags)) {
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags)) {
- 		rtl_drop_queued_tx(tp);
- 		return;
- 	}
-@@ -6338,7 +6338,7 @@ static void rtl8153b_up(struct r8152 *tp
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153b_u1u2en(tp, false);
-@@ -6362,7 +6362,7 @@ static void rtl8153b_down(struct r8152 *
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags)) {
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags)) {
- 		rtl_drop_queued_tx(tp);
- 		return;
- 	}
-@@ -6399,7 +6399,7 @@ static void rtl8153c_up(struct r8152 *tp
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153b_u1u2en(tp, false);
-@@ -6480,7 +6480,7 @@ static void rtl8156_up(struct r8152 *tp)
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153b_u1u2en(tp, false);
-@@ -6553,7 +6553,7 @@ static void rtl8156_down(struct r8152 *t
- {
- 	u32 ocp_data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags)) {
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags)) {
- 		rtl_drop_queued_tx(tp);
- 		return;
- 	}
-@@ -6691,7 +6691,7 @@ static void rtl_work_func_t(struct work_
- 	/* If the device is unplugged or !netif_running(), the workqueue
- 	 * doesn't need to wake the device, and could return directly.
- 	 */
--	if (test_bit(RTL8152_UNPLUG, &tp->flags) || !netif_running(tp->netdev))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags) || !netif_running(tp->netdev))
- 		return;
- 
- 	if (usb_autopm_get_interface(tp->intf) < 0)
-@@ -6730,7 +6730,7 @@ static void rtl_hw_phy_work_func_t(struc
- {
- 	struct r8152 *tp = container_of(work, struct r8152, hw_phy_work.work);
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	if (usb_autopm_get_interface(tp->intf) < 0)
-@@ -6857,7 +6857,7 @@ static int rtl8152_close(struct net_devi
- 	netif_stop_queue(netdev);
- 
- 	res = usb_autopm_get_interface(tp->intf);
--	if (res < 0 || test_bit(RTL8152_UNPLUG, &tp->flags)) {
-+	if (res < 0 || test_bit(RTL8152_INACCESSIBLE, &tp->flags)) {
- 		rtl_drop_queued_tx(tp);
- 		rtl_stop_rx(tp);
- 	} else {
-@@ -6890,7 +6890,7 @@ static void r8152b_init(struct r8152 *tp
- 	u32 ocp_data;
- 	u16 data;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	data = r8152_mdio_read(tp, MII_BMCR);
-@@ -6934,7 +6934,7 @@ static void r8153_init(struct r8152 *tp)
- 	u16 data;
- 	int i;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153_u1u2en(tp, false);
-@@ -6945,7 +6945,7 @@ static void r8153_init(struct r8152 *tp)
- 			break;
- 
- 		msleep(20);
--		if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+		if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 			break;
- 	}
- 
-@@ -7074,7 +7074,7 @@ static void r8153b_init(struct r8152 *tp
- 	u16 data;
- 	int i;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153b_u1u2en(tp, false);
-@@ -7085,7 +7085,7 @@ static void r8153b_init(struct r8152 *tp
- 			break;
- 
- 		msleep(20);
--		if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+		if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 			break;
- 	}
- 
-@@ -7156,7 +7156,7 @@ static void r8153c_init(struct r8152 *tp
- 	u16 data;
- 	int i;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153b_u1u2en(tp, false);
-@@ -7176,7 +7176,7 @@ static void r8153c_init(struct r8152 *tp
- 			break;
- 
- 		msleep(20);
--		if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+		if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 			return;
- 	}
- 
-@@ -8005,7 +8005,7 @@ static void r8156_init(struct r8152 *tp)
- 	u16 data;
- 	int i;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	ocp_data = ocp_read_byte(tp, MCU_TYPE_USB, USB_ECM_OP);
-@@ -8026,7 +8026,7 @@ static void r8156_init(struct r8152 *tp)
- 			break;
- 
- 		msleep(20);
--		if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+		if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 			return;
- 	}
- 
-@@ -8101,7 +8101,7 @@ static void r8156b_init(struct r8152 *tp
- 	u16 data;
- 	int i;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	ocp_data = ocp_read_byte(tp, MCU_TYPE_USB, USB_ECM_OP);
-@@ -8135,7 +8135,7 @@ static void r8156b_init(struct r8152 *tp
- 			break;
- 
- 		msleep(20);
--		if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+		if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 			return;
- 	}
- 
-@@ -9164,7 +9164,7 @@ static int rtl8152_ioctl(struct net_devi
- 	struct mii_ioctl_data *data = if_mii(rq);
- 	int res;
- 
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return -ENODEV;
- 
- 	res = usb_autopm_get_interface(tp->intf);
-@@ -9266,7 +9266,7 @@ static const struct net_device_ops rtl81
- 
- static void rtl8152_unload(struct r8152 *tp)
- {
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	if (tp->version != RTL_VER_01)
-@@ -9275,7 +9275,7 @@ static void rtl8152_unload(struct r8152
- 
- static void rtl8153_unload(struct r8152 *tp)
- {
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153_power_cut_en(tp, false);
-@@ -9283,7 +9283,7 @@ static void rtl8153_unload(struct r8152
- 
- static void rtl8153b_unload(struct r8152 *tp)
- {
--	if (test_bit(RTL8152_UNPLUG, &tp->flags))
-+	if (test_bit(RTL8152_INACCESSIBLE, &tp->flags))
- 		return;
- 
- 	r8153b_power_cut_en(tp, false);
diff --git a/target/linux/generic/backport-6.1/795-v6.6-13-r8152-Block-future-register-access-if-register-acces.patch b/target/linux/generic/backport-6.1/795-v6.6-13-r8152-Block-future-register-access-if-register-acces.patch
index 2cbe3650352d6..8901767be5239 100644
--- a/target/linux/generic/backport-6.1/795-v6.6-13-r8152-Block-future-register-access-if-register-acces.patch
+++ b/target/linux/generic/backport-6.1/795-v6.6-13-r8152-Block-future-register-access-if-register-acces.patch
@@ -232,7 +232,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  }
  
  static int generic_ocp_read(struct r8152 *tp, u16 index, u16 size,
-@@ -8261,7 +8355,7 @@ static int rtl8152_pre_reset(struct usb_
+@@ -8275,7 +8369,7 @@ static int rtl8152_pre_reset(struct usb_
  	struct r8152 *tp = usb_get_intfdata(intf);
  	struct net_device *netdev;
  
@@ -241,7 +241,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		return 0;
  
  	netdev = tp->netdev;
-@@ -8276,7 +8370,9 @@ static int rtl8152_pre_reset(struct usb_
+@@ -8290,7 +8384,9 @@ static int rtl8152_pre_reset(struct usb_
  	napi_disable(&tp->napi);
  	if (netif_carrier_ok(netdev)) {
  		mutex_lock(&tp->control);
@@ -251,7 +251,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		mutex_unlock(&tp->control);
  	}
  
-@@ -8289,9 +8385,11 @@ static int rtl8152_post_reset(struct usb
+@@ -8303,9 +8399,11 @@ static int rtl8152_post_reset(struct usb
  	struct net_device *netdev;
  	struct sockaddr sa;
  
@@ -264,7 +264,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	/* reset the MAC address in case of policy change */
  	if (determine_ethernet_addr(tp, &sa) >= 0) {
  		rtnl_lock();
-@@ -9493,17 +9591,29 @@ static u8 __rtl_get_hw_ver(struct usb_de
+@@ -9507,17 +9605,29 @@ static u8 __rtl_get_hw_ver(struct usb_de
  	__le32 *tmp;
  	u8 version;
  	int ret;
@@ -300,7 +300,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  
  	kfree(tmp);
  
-@@ -9602,25 +9712,14 @@ static bool rtl8152_supports_lenovo_macp
+@@ -9616,25 +9726,14 @@ static bool rtl8152_supports_lenovo_macp
  	return 0;
  }
  
@@ -328,7 +328,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	usb_reset_device(udev);
  	netdev = alloc_etherdev(sizeof(struct r8152));
  	if (!netdev) {
-@@ -9783,10 +9882,20 @@ static int rtl8152_probe(struct usb_inte
+@@ -9797,10 +9896,20 @@ static int rtl8152_probe(struct usb_inte
  	else
  		device_set_wakeup_enable(&udev->dev, false);
  
@@ -349,7 +349,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  out1:
  	tasklet_kill(&tp->tx_tl);
  	cancel_delayed_work_sync(&tp->hw_phy_work);
-@@ -9795,10 +9904,46 @@ out1:
+@@ -9809,10 +9918,46 @@ out1:
  	rtl8152_release_firmware(tp);
  	usb_set_intfdata(intf, NULL);
  out:
diff --git a/target/linux/generic/hack-6.1/760-net-usb-r8152-add-LED-configuration-from-OF.patch b/target/linux/generic/hack-6.1/760-net-usb-r8152-add-LED-configuration-from-OF.patch
index c842639792f12..48d4626ed632e 100644
--- a/target/linux/generic/hack-6.1/760-net-usb-r8152-add-LED-configuration-from-OF.patch
+++ b/target/linux/generic/hack-6.1/760-net-usb-r8152-add-LED-configuration-from-OF.patch
@@ -22,7 +22,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  #include <linux/crc32.h>
  #include <linux/if_vlan.h>
  #include <linux/uaccess.h>
-@@ -7020,6 +7021,22 @@ static void rtl_tally_reset(struct r8152
+@@ -7034,6 +7035,22 @@ static void rtl_tally_reset(struct r8152
  	ocp_write_word(tp, MCU_TYPE_PLA, PLA_RSTTALLY, ocp_data);
  }
  
@@ -45,7 +45,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  static void r8152b_init(struct r8152 *tp)
  {
  	u32 ocp_data;
-@@ -7061,6 +7078,8 @@ static void r8152b_init(struct r8152 *tp
+@@ -7075,6 +7092,8 @@ static void r8152b_init(struct r8152 *tp
  	ocp_data = ocp_read_word(tp, MCU_TYPE_USB, USB_USB_CTRL);
  	ocp_data &= ~(RX_AGG_DISABLE | RX_ZERO_EN);
  	ocp_write_word(tp, MCU_TYPE_USB, USB_USB_CTRL, ocp_data);
@@ -54,7 +54,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  }
  
  static void r8153_init(struct r8152 *tp)
-@@ -7201,6 +7220,8 @@ static void r8153_init(struct r8152 *tp)
+@@ -7215,6 +7234,8 @@ static void r8153_init(struct r8152 *tp)
  		tp->coalesce = COALESCE_SLOW;
  		break;
  	}
@@ -63,7 +63,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  }
  
  static void r8153b_init(struct r8152 *tp)
-@@ -7283,6 +7304,8 @@ static void r8153b_init(struct r8152 *tp
+@@ -7297,6 +7318,8 @@ static void r8153b_init(struct r8152 *tp
  	rtl_tally_reset(tp);
  
  	tp->coalesce = 15000;	/* 15 us */
diff --git a/target/linux/generic/hack-6.1/800-GPIO-add-named-gpio-exports.patch b/target/linux/generic/hack-6.1/800-GPIO-add-named-gpio-exports.patch
index ba5351275dd0b..b60b428dc8a83 100644
--- a/target/linux/generic/hack-6.1/800-GPIO-add-named-gpio-exports.patch
+++ b/target/linux/generic/hack-6.1/800-GPIO-add-named-gpio-exports.patch
@@ -157,7 +157,7 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
  {
 --- a/drivers/gpio/gpiolib-sysfs.c
 +++ b/drivers/gpio/gpiolib-sysfs.c
-@@ -544,7 +544,7 @@ static struct class gpio_class = {
+@@ -547,7 +547,7 @@ static struct class gpio_class = {
   *
   * Returns zero on success, else an error.
   */
@@ -166,7 +166,7 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
  {
  	struct gpio_chip	*chip;
  	struct gpio_device	*gdev;
-@@ -606,6 +606,8 @@ int gpiod_export(struct gpio_desc *desc,
+@@ -609,6 +609,8 @@ int gpiod_export(struct gpio_desc *desc,
  	offset = gpio_chip_hwgpio(desc);
  	if (chip->names && chip->names[offset])
  		ioname = chip->names[offset];
@@ -175,7 +175,7 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
  
  	dev = device_create_with_groups(&gpio_class, &gdev->dev,
  					MKDEV(0, 0), data, gpio_groups,
-@@ -627,6 +629,12 @@ err_unlock:
+@@ -630,6 +632,12 @@ err_unlock:
  	gpiod_dbg(desc, "%s: status %d\n", __func__, status);
  	return status;
  }
diff --git a/target/linux/generic/pending-6.1/102-MIPS-only-process-negative-stack-offsets-on-stack-tr.patch b/target/linux/generic/pending-6.1/102-MIPS-only-process-negative-stack-offsets-on-stack-tr.patch
index 7ce3d491cd327..d79d03defb3ad 100644
--- a/target/linux/generic/pending-6.1/102-MIPS-only-process-negative-stack-offsets-on-stack-tr.patch
+++ b/target/linux/generic/pending-6.1/102-MIPS-only-process-negative-stack-offsets-on-stack-tr.patch
@@ -46,7 +46,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 
 --- a/arch/mips/kernel/process.c
 +++ b/arch/mips/kernel/process.c
-@@ -394,6 +394,8 @@ static inline int is_sp_move_ins(union m
+@@ -395,6 +395,8 @@ static inline int is_sp_move_ins(union m
  
  	if (ip->i_format.opcode == addiu_op ||
  	    ip->i_format.opcode == daddiu_op) {
diff --git a/target/linux/generic/pending-6.1/300-mips_expose_boot_raw.patch b/target/linux/generic/pending-6.1/300-mips_expose_boot_raw.patch
index 9e571a6d875b1..d222ec060e8b7 100644
--- a/target/linux/generic/pending-6.1/300-mips_expose_boot_raw.patch
+++ b/target/linux/generic/pending-6.1/300-mips_expose_boot_raw.patch
@@ -9,7 +9,7 @@ Acked-by: Rob Landley <rob@landley.net>
 ---
 --- a/arch/mips/Kconfig
 +++ b/arch/mips/Kconfig
-@@ -1034,9 +1034,6 @@ config FW_ARC
+@@ -1035,9 +1035,6 @@ config FW_ARC
  config ARCH_MAY_HAVE_PC_FDC
  	bool
  
@@ -19,7 +19,7 @@ Acked-by: Rob Landley <rob@landley.net>
  config CEVT_BCM1480
  	bool
  
-@@ -3091,6 +3088,18 @@ choice
+@@ -3093,6 +3090,18 @@ choice
  		bool "Extend builtin kernel arguments with bootloader arguments"
  endchoice
  
diff --git a/target/linux/generic/pending-6.1/701-netfilter-nf_tables-ignore-EOPNOTSUPP-on-flowtable-d.patch b/target/linux/generic/pending-6.1/701-netfilter-nf_tables-ignore-EOPNOTSUPP-on-flowtable-d.patch
index 8c508e3657963..307ddce761fc1 100644
--- a/target/linux/generic/pending-6.1/701-netfilter-nf_tables-ignore-EOPNOTSUPP-on-flowtable-d.patch
+++ b/target/linux/generic/pending-6.1/701-netfilter-nf_tables-ignore-EOPNOTSUPP-on-flowtable-d.patch
@@ -18,7 +18,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 
 --- a/net/netfilter/nf_tables_api.c
 +++ b/net/netfilter/nf_tables_api.c
-@@ -7883,7 +7883,7 @@ static int nft_register_flowtable_net_ho
+@@ -7884,7 +7884,7 @@ static int nft_register_flowtable_net_ho
  		err = flowtable->data.type->setup(&flowtable->data,
  						  hook->ops.dev,
  						  FLOW_BLOCK_BIND);
diff --git a/target/linux/ipq806x/patches-6.1/700-02-net-stmmac-move-TX-timer-arm-after-DMA-enable.patch b/target/linux/ipq806x/patches-6.1/700-02-net-stmmac-move-TX-timer-arm-after-DMA-enable.patch
index 085d38a79a777..222aa3f779c4b 100644
--- a/target/linux/ipq806x/patches-6.1/700-02-net-stmmac-move-TX-timer-arm-after-DMA-enable.patch
+++ b/target/linux/ipq806x/patches-6.1/700-02-net-stmmac-move-TX-timer-arm-after-DMA-enable.patch
@@ -17,7 +17,7 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
 
 --- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
 +++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
-@@ -2529,7 +2529,8 @@ static void stmmac_bump_dma_threshold(st
+@@ -2530,7 +2530,8 @@ static void stmmac_bump_dma_threshold(st
   * @queue: TX queue index
   * Description: it reclaims the transmit resources after transmission completes.
   */
@@ -27,7 +27,7 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
  {
  	struct stmmac_tx_queue *tx_q = &priv->dma_conf.tx_queue[queue];
  	unsigned int bytes_compl = 0, pkts_compl = 0;
-@@ -2692,7 +2693,7 @@ static int stmmac_tx_clean(struct stmmac
+@@ -2693,7 +2694,7 @@ static int stmmac_tx_clean(struct stmmac
  
  	/* We still have pending packets, let's call for a new scheduling */
  	if (tx_q->dirty_tx != tx_q->cur_tx)
@@ -36,7 +36,7 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
  
  	__netif_tx_unlock_bh(netdev_get_tx_queue(priv->dev, queue));
  
-@@ -5475,12 +5476,13 @@ static int stmmac_napi_poll_tx(struct na
+@@ -5476,12 +5477,13 @@ static int stmmac_napi_poll_tx(struct na
  	struct stmmac_channel *ch =
  		container_of(napi, struct stmmac_channel, tx_napi);
  	struct stmmac_priv *priv = ch->priv_data;
@@ -51,7 +51,7 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
  	work_done = min(work_done, budget);
  
  	if (work_done < budget && napi_complete_done(napi, work_done)) {
-@@ -5491,6 +5493,10 @@ static int stmmac_napi_poll_tx(struct na
+@@ -5492,6 +5494,10 @@ static int stmmac_napi_poll_tx(struct na
  		spin_unlock_irqrestore(&ch->lock, flags);
  	}
  
@@ -62,7 +62,7 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
  	return work_done;
  }
  
-@@ -5500,11 +5506,12 @@ static int stmmac_napi_poll_rxtx(struct
+@@ -5501,11 +5507,12 @@ static int stmmac_napi_poll_rxtx(struct
  		container_of(napi, struct stmmac_channel, rxtx_napi);
  	struct stmmac_priv *priv = ch->priv_data;
  	int rx_done, tx_done, rxtx_done;
@@ -76,7 +76,7 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
  	tx_done = min(tx_done, budget);
  
  	rx_done = stmmac_rx_zc(priv, budget, chan);
-@@ -5529,6 +5536,10 @@ static int stmmac_napi_poll_rxtx(struct
+@@ -5530,6 +5537,10 @@ static int stmmac_napi_poll_rxtx(struct
  		spin_unlock_irqrestore(&ch->lock, flags);
  	}
  
diff --git a/target/linux/ipq806x/patches-6.1/700-03-net-stmmac-improve-TX-timer-arm-logic.patch b/target/linux/ipq806x/patches-6.1/700-03-net-stmmac-improve-TX-timer-arm-logic.patch
index 6160b95e605b2..21c7ff1777d3f 100644
--- a/target/linux/ipq806x/patches-6.1/700-03-net-stmmac-improve-TX-timer-arm-logic.patch
+++ b/target/linux/ipq806x/patches-6.1/700-03-net-stmmac-improve-TX-timer-arm-logic.patch
@@ -45,7 +45,7 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
 
 --- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
 +++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
-@@ -2975,13 +2975,25 @@ static void stmmac_tx_timer_arm(struct s
+@@ -2976,13 +2976,25 @@ static void stmmac_tx_timer_arm(struct s
  {
  	struct stmmac_tx_queue *tx_q = &priv->dma_conf.tx_queue[queue];
  	u32 tx_coal_timer = priv->tx_coal_timer[queue];
diff --git a/target/linux/mediatek/patches-6.1/100-dts-update-mt7622-rfb1.patch b/target/linux/mediatek/patches-6.1/100-dts-update-mt7622-rfb1.patch
index 1c249c03a6848..1afbcaf03853a 100644
--- a/target/linux/mediatek/patches-6.1/100-dts-update-mt7622-rfb1.patch
+++ b/target/linux/mediatek/patches-6.1/100-dts-update-mt7622-rfb1.patch
@@ -35,8 +35,9 @@
  		};
  	};
  
- 	memory {
+-	memory@40000000 {
 -		reg = <0 0x40000000 0 0x20000000>;
++	memory {
 +		reg = <0 0x40000000 0 0x40000000>;
  	};
  
diff --git a/target/linux/octeon/patches-6.1/100-mips_image_cmdline_hack.patch b/target/linux/octeon/patches-6.1/100-mips_image_cmdline_hack.patch
index 08f30c5e03f16..f63af640dde95 100644
--- a/target/linux/octeon/patches-6.1/100-mips_image_cmdline_hack.patch
+++ b/target/linux/octeon/patches-6.1/100-mips_image_cmdline_hack.patch
@@ -10,7 +10,7 @@ Signed-off-by: Gabor Juhos <juhosg@openwrt.org>
 
 --- a/arch/mips/Kconfig
 +++ b/arch/mips/Kconfig
-@@ -1114,6 +1114,10 @@ config MIPS_MSC
+@@ -1115,6 +1115,10 @@ config MIPS_MSC
  config SYNC_R4K
  	bool
  
